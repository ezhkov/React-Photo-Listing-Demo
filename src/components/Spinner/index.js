import React from 'react';
import './Spinner.css';

const Spinner = () => (
  <div className="site-spinner">
    <div className="lds-ring">
      <div>&nbsp;</div>
      <div>&nbsp;</div>
      <div>&nbsp;</div>
      <div>&nbsp;</div>
    </div>
  </div>
);

export default Spinner;
