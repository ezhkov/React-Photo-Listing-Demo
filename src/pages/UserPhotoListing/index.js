import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { Row, Col, Pagination } from 'antd';
import axios from 'axios';
import { Spinner, PhotoCard } from '../../components';
import { URL_FOR_USER_PHOTO_LISTING_QUERY } from '../../constants/urls';
import './index.css';

class UserPhotoListing extends Component {
  constructor(...args) {
    super(...args);
    this.state = {
      isListingLoading: false,
      cards: [],
      totalCards: 10,
      page: 1,
      per_page: 6,
    };
  }

  componentDidMount = () => {
    this.handleUserPhotoListingQuery();
  };

  handlePaginationChange = (current) => {
    this.setState({
      page: current,
      per_page: 6,
    }, this.handleUserPhotoListingQuery);
  };

  handleUserPhotoListingQuery = () => {
    const { history, userId } = this.props;
    const { page, per_page: perPage } = this.state;
    this.setState({ isListingLoading: true });
    const API_URL = URL_FOR_USER_PHOTO_LISTING_QUERY(userId);
    axios.get(API_URL, {
      params: {
        page,
        perPage,
        client_id: process.env.REACT_APP_UNSPLASH_API_KEY,
      },
    }).then((res) => {
      const cards = res.data;
      const totalCards = parseInt(res.headers['x-total'], 10);
      this.setState({
        cards,
        isListingLoading: false,
        totalCards,
      });
    })
      .catch(() => {
        history.push('/');
      });
  }

  render() {
    const {
      isListingLoading,
      cards,
      totalCards,
      page,
      per_page: perPage,
    } = this.state;
    return (
      <div>
        { isListingLoading && (<Spinner className="spinner" />)}
        <div>
          <Row justify="center" style={{ margin: '20px 0' }}>
            <Col span={24}>
              {!isListingLoading && (
                <ul className="photo-list user-photos ">
                  {
                    cards.map(item => (
                      <li key={item.id} className="photo-list__item pl-3">
                        <PhotoCard
                          photoName={item.urls.regular}
                          title=""
                          tags={item.photo_tags}
                          photoID={item.id}
                          userID={item.user.username}
                          userAvatar={item.user.profile_image.small}
                          onSearchTagValue={this.handleSearchText}
                        />
                      </li>
                    ))
                  }
                </ul>
              )}
            </Col>
          </Row>
          <Row justify="center" style={{ display: 'flex', justifyContent: 'center' }}>
            <Col style={{ display: 'flex', justifyContent: 'center' }}>
              {totalCards > perPage && (
                <Pagination
                  className="ml-3 mb-5"
                  onChange={this.handlePaginationChange}
                  showSizeChanger
                  current={page}
                  defaultCurrent={1}
                  total={totalCards}
                />
              )}
            </Col>
          </Row>
        </div>
      </div>
    );
  }
}

UserPhotoListing.propTypes = {
  history: PropTypes.shape({
    prop: PropTypes.string,
  }),
  userId: PropTypes.string,
};
UserPhotoListing.defaultProps = {
  history: {},
  userId: '',
};
export default UserPhotoListing;
