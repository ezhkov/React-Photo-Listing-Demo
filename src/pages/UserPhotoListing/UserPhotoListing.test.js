import React from 'react';
import Enzyme, { shallow, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import UserPhotoListing from '.'; 
import Spinner from '../../components/Spinner';

Enzyme.configure({ adapter: new Adapter() });

describe('Test of component of UserPhotoListing', () => {
    it('Test display Pagination', () => { 
        const props = {
            userId: 'sC-BXbi9ajw'
        }

        const userPhotoListing = shallow(<UserPhotoListing {...props} />);

        userPhotoListing.setState({ 
            totalCards: 7,
          }); 

        expect(userPhotoListing.find('Pagination').length).toEqual(1);

        userPhotoListing.setState({ 
            totalCards: 5,
          }); 

        expect(userPhotoListing.find('Pagination').length).toEqual(0);
        //expect(userStatistic.find('.user-statistic__chart').at(1).prop('config')).toEqual(undefined);
    });  

    it('Test Spinner', () => { 
        const props = {
            userId: 'sC-BXbi9ajw'
        }

        const userPhotoListing = shallow(<UserPhotoListing {...props} />);
        
        
        userPhotoListing.setState({ 
          isListingLoading: false,
        }); 
        expect(userPhotoListing.find(Spinner).length).toEqual(0);
    
        userPhotoListing.setState({ 
          isListingLoading: true,
        }); 
    
        expect(userPhotoListing.find(Spinner).length).toEqual(1);
        
    });

    it('Test Snapshot', () => { 
        const props = {
            userId: 'sC-BXbi9ajw'
        }

        const userPhotoListing = shallow(<UserPhotoListing {...props} />);
        expect(userPhotoListing).toMatchSnapshot();
    });
});