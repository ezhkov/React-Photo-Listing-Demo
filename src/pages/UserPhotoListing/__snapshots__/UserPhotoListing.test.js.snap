// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test of component of UserPhotoListing Test Snapshot 1`] = `ShallowWrapper {}`;
